### PR TITLE
Fixes for order and missing terms in trigonometric integrals

### LINKED ISSUE #26207: Asymptotic series expansion for Cosine Integral function
I was just testing out the asymptotic series & limits at infinity for some error functions and I found this 
```
>>> Si(x).series(x, oo)
pi/2 - (-6/x**3 + 1/x + O(x**(-7), (x, oo)))*sin(x)/x - (24/x**4 - 2/x**2 + 1 + O(x**(-7), (x, oo)))*cos(x)/x
>>>
>>> Si(x).series(x, -oo)
-24*cos(x)/x**5 + 6*sin(x)/x**4 + 2*cos(x)/x**3 - sin(x)/x**2 - cos(x)/x - pi/2 + O(x**(-6), (x, -oo))
>>>
>>> Ci(x).series(x, oo)
-(-6/x**3 + 1/x + O(x**(-7), (x, oo)))*cos(x)/x + (24/x**4 - 2/x**2 + 1 + O(x**(-7), (x, oo)))*sin(x)/x
>>>
>>> Ci(x).series(x, -oo)
Traceback (most recent call last):
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 2994, in series
    s = self.subs(x, cdir/x).series(x, n=n, dir='+', cdir=1)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3015, in series
    rv = self.subs(x, xpos).series(xpos, x0, n, dir, logx=logx, cdir=cdir)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3023, in series
    s1 = self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 1800, in _eval_nseries
    return super()._eval_nseries(x, n, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 690, in _eval_nseries
    return self._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 2082, in _eval_aseries
    return super(Ci, self)._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 653, in _eval_aseries
    raise PoleError(filldedent('''
sympy.core.function.PoleError:
Asymptotic expansion of Ci around [-oo] is not implemented.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 2999, in series
    s = self.subs(x, cdir*x).aseries(x, n=n)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3198, in aseries
    return self.subs(x, xpos).aseries(xpos, n, bound, hir).subs(xpos, x)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3216, in aseries
    s = self.subs(x, exp(x)).aseries(x, n, bound, hir).subs(x, log(x))
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3238, in aseries
    s = func.series(k, 0, n)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3023, in series
    s1 = self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 1800, in _eval_nseries
    return super()._eval_nseries(x, n, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 690, in _eval_nseries
    return self._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 2082, in _eval_aseries
    return super(Ci, self)._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 653, in _eval_aseries
    raise PoleError(filldedent('''
sympy.core.function.PoleError:
Asymptotic expansion of Ci around [-oo] is not implemented.
```

### LINKED ISSUE #26257: Incorrect result while expanding `Ci(x)` around -infinty.
I got this result while expanding `Ci(x)` for -inf
```
>> Ci(x).series(x,-oo)
I*pi - (-6/x**3 + 1/x + O(-1/x**7, (x, -oo)))*cos(x)/x + (24/x**4 - 2/x**2 + 1 + O(-1/x**7, (x, -oo)))*sin(x)/x
```
while on wolfram it has an extra O(-1/x**7) term with it.
image

### PR DESCRIPTION
**Pull request by arnabnandikgp**
#### References to other Issues or PRs
Fixes #26207
Fixes #26257

#### Release Notes

* functions
  * Fixes for order and missing terms in trigonometric integrals in error functions

### CODE REVIEW

File: sympy/functions/special/tests/test_error_functions.py [hunk 1]
```diff
@@ -675,16 +675,16 @@ def test_ci():
...
     assert Chi(x).nseries(x, n=4) == \
-        EulerGamma + log(x) + x**2/4 + x**4/96 + O(x**5)
+        EulerGamma + log(x) + x**2/4 + O(x**4)
 
-    assert Ci(x).series(x, oo) == -cos(x)*(-6/x**3 + 1/x \
-        + O(x**(-7), (x, oo)))/x + (24/x**4 - 2/x**2 + 1 \
-        + O(x**(-7), (x, oo)))*sin(x)/x
+    assert Ci(x).series(x, oo) == -cos(x)*(120/x**5 -6/x**3 + 1/x \
+        + O(x**(-6), (x, oo)))/x + (24/x**4 - 2/x**2 + 1 \
+        + O(x**(-6), (x, oo)))*sin(x)/x
```
[the remark this answers is not in the document]
Comment (anutosh491) [reply sympy/functions/special/tests/test_error_functions.py:684]: When simplified to a single order term, we would expect something like what we have for `sin`
```
>>> sin(1/x).series(x, oo)
1/(120*x**5) - 1/(6*x**3) + 1/x + O(x**(-6), (x, oo))
```
The order term should have a `-6` power

[the remark this answers is not in the document]
Comment (anutosh491) [reply sympy/functions/special/tests/test_error_functions.py:684]: Maybe check comment (https://github.com/sympy/sympy/issues/26257#issuecomment-1962331682) and the one below on the issue

[the remark this answers is not in the document]
Comment (arnabnandikgp) [reply sympy/functions/special/tests/test_error_functions.py:684]: Thanks for reviewing. After reading your comments, I think I should state the following in order to sort any confusions regarding this issue for me:
1. We agree on the fact that for a standalone series like `...+ 120/x**5 -6/x**3 + 1/x ` the order term should ofcourse be `O(x**(-6), (x, oo))` (if only the above three terms have been printed till now). Now the problem with trigonometric integrals here is that we have such series multiplied with terms like `cos(x)/x` or `sin(x)/x` so after we use simplification tools like `expand()` sympy detects  a infinite series with now having the largest(magnitude) term of x**(-6) and hence corrects it to an order term of x**(-7) , which we do not want as a response for a simplified form for an expression like `Ci(x).series(x, oo, 6)` .
So basically for the sanity of `O(x**(-6), (x, oo)) ` going with `...+ 120/x**5 -6/x**3 + 1/x ` we lose integrity for the response of `Ci(x).series(x, oo, 6)` being computed upto 6th order , so the best way out here will be to maybe return the expanded of the series first hand in place of addition of two different infinite series:
the following diff on top of the current pr branch may solve this:
```diff
--- a/sympy/functions/special/error_functions.py
+++ b/sympy/functions/special/error_functions.py
@@ -2071,10 +2071,10 @@ def _eval_aseries(self, n, args0, x, logx):
         if point in (S.Infinity, S.NegativeInfinity):
             z = self.args[0]
             p = [S.NegativeOne**k * factorial(2*k) / z**(2*k)
-                    for k in range(int(n/2)+1)] + [Order(1/z**n, x)]
+                    for k in range(int(n/2)+1)]
             q = [S.NegativeOne**k * factorial(2*k + 1) / z**(2*k + 1)
-                    for k in range(int(n/2))] + [Order(1/z**n, x)]
-            result = (sin(z)/z)*Add(*p) - (cos(z)/z)*Add(*q)
+                    for k in range(int(n/2))] 
+            result = expand_mul((sin(z)/z)*Add(*p) - (cos(z)/z)*Add(*q)) + Order(1/z**n, x)
 
             if point is S.NegativeInfinity:

```
which will directly yield 
```
In [6]: Ci(x).series(x, oo, 6)
Out[6]: 
24⋅sin(x)   6⋅cos(x)   2⋅sin(x)   cos(x)   sin(x)    ⎛1        ⎞
───────── + ──────── - ──────── - ────── + ────── + O⎜──; x → ∞⎟
    5           4          3         2       x       ⎜ 6       ⎟
   x           x          x         x                ⎝x        ⎠

```

[the remark this answers is not in the document]
Comment (anutosh491) [reply sympy/functions/special/tests/test_error_functions.py:684]: I would say let's stick to the output given by wolfram while preserving the order (to -6 in general ) if that's possible. Could you maybe look up for references or wolfram and check the result there ?

[the remark this answers is not in the document]
Comment (anutosh491) [reply sympy/functions/special/tests/test_error_functions.py]: Perfect yes that would be something I would like to have. Have you verified it with wolfram and tested relevant examples on sympy. If yes I think we can go ahead with merging this (I'll review it once you reply)

[the remark this answers is not in the document]
Comment (arnabnandikgp) [reply sympy/functions/special/tests/test_error_functions.py]: Yes, I have verified the test cases and some other examples with the results from wolfram

File: sympy/functions/special/error_functions.py [hunk 2]
```diff
@@ -1928,11 +1927,11 @@ def _eval_aseries(self, n, args0, x, logx):
         # Expansion at oo
         if point is S.Infinity:
             z = self.args[0]
-            p = [S.NegativeOne**k * factorial(2*k) / z**(2*k)
-                    for k in range(int((n - 1)/2))] + [Order(1/z**n, x)]
-            q = [S.NegativeOne**k * factorial(2*k + 1) / z**(2*k + 1)
-                    for k in range(int(n/2) - 1)] + [Order(1/z**n, x)]
-            return pi/2 - (cos(z)/z)*Add(*p) - (sin(z)/z)*Add(*q)
+            p = [S.NegativeOne**k * factorial(2*k) / z**(2*k + 1)
+                    for k in range(int(n/2) + 1)] + [Order(1/z**n, x)]
```
Comment (oscarbenjamin) [inline sympy/functions/special/error_functions.py:1931]: Rather than `int(n/2)` which uses floating point arithmetic it would be better to use `n // 2`.

**Review by anutosh491 — APPROVED** [review]
Few tests fail but I don't think they are arising from the changes made here. Looks good.

**Comment by arnabnandikgp** [comment]
I think the failing numpy job is passing now, we can restart the tests if needed before merging.

**Comment by oscarbenjamin** [comment]
> think the failing numpy job is passing now, we can restart the tests if needed before merging.

It doesn't matter if the numpy nightly job fails. The PR can still be merged anyway and there is no need to restart the tests.

**Pull request merged**